test/helper/sending.bash

```diff
@@ -82,20 +132,35 @@ function _send_email() {
 # chosen. Sending more than one mail at any given point in time with this function
 # is UNDEFINED BEHAVIOR!
 function _send_email_and_get_id() {
-  [[ -v CONTAINER_NAME ]] || return 1
-
-  _wait_for_empty_mail_queue_in_container
-  _send_email "${@}"
-  _wait_for_empty_mail_queue_in_container
+  # Export the variable denoted by ${1} so everyone has access
+  export "${1:?Mail ID must be set for _send_email_and_get_id}"
+  # Get a "reference" to the content of the variable denoted by ${1} so we can manipulate the content
+  local -n ID_ENV_VAR_REF=${1:?}
+  # Prepare the message ID header here because we will shift away ${1} later
+  local MID="<${1}@dms-tests>"
+  # Get rid of ${1} so only the arguments for swaks remain
+  shift 1
 
-  local MAIL_ID
+  local QUEUE_ID
   # The unique ID Postfix (and other services) use may be different in length
   # on different systems (e.g. amd64 (11) vs aarch64 (10)). Hence, we use a
   # range to safely capture it.
-  MAIL_ID=$(_exec_in_container tac /var/log/mail.log              \
-    | grep -E -m 1 'postfix/smtpd.*: [A-Z0-9]+: client=localhost' \
-    | grep -E -o '[A-Z0-9]{9,12}' || true)
+  local QUEUE_ID_REGEX='[A-Z0-9]{9,12}'
+
+  _wait_for_empty_mail_queue_in_container
+  local OUTPUT=$(_send_email "${@}" --header "Message-Id: ${MID}")
+  _wait_for_empty_mail_queue_in_container
```

Comment:
`OUTPUT` doesn't seem to be used anymore?

You could also technically avoid the initial queue empty condition, as you're providing an explicit message ID to match against in logs you're only interested in the queue being empty after the mail was sent?

Only concern to have both waits is postfix is configured for short IDs encoding, and this has a gotcha of rare chance the Queue ID is reused apparently. I'm not quite sure why since it's not only the microseconds encoded, but also the inode number 🤷‍♂️ 

The long queue IDs encode more information where they're unique which can avoid that.